### DISCUSSION TITLE
Security Verification Failures (nginx config?) on 4.3 Docker migration

### DISCUSSION THREAD

**Discussion by tr0043t** [mastodon/mastodon#32566] [Q&A]
I'm in the process of migrating from a 4.2.x non-docker build to a 4.3.x docker build. I'm at the point where the docker build comes up as a test site internally, having brought in the database from the 4.2 prod system. However, I'm hitting authentication errors that are real blockers.

One is that my own Owner account keeps asking for 2fa but refuses to accept it, and also rejects my backup codes. I used tootctl to (in theory) remove 2fa, but it still is being demanded.

Of even more concern is that when I created a new account (using tootctl) I can't log it in because I keep getting:

"Security verification failed, did you block cookies?" errors.

Research suggests this is a recurring problem over the years, and some sources point to header/origin issues in the nginx reverse proxy configuration, but after trying a variety of approaches there (and also obviously including wiping out cookies and caches and such) I'm stumped.

Any suggestions greatly appreciated. Thanks!

**Comment by mjankowski** [comment]
Did you sort any of that out?

Sounds like maybe related but separate issues.

For the 2FA portion, there was a change in the 4.2 to 4.3 upgrade ... did you follow those steps? (as long as the values were previously working, they should have kept working ... there would be an error during migration if they were previously broken/changed by accident).

For tootctl thing -- if you can replicate this (a command not doing what it should), can you open issue just for that?
